eng/pipelines/templates/stages/vmr-verticals.yml

```diff
@@ -995,3 +995,77 @@ stages:
             - Windows_x64
             - Windows_x86
             - Windows_arm64
+
+        # Source Build Leg
+        - template: ../jobs/vmr-build.yml
+          parameters:
+            buildName: ${{ format('{0}_SourceBuild_Online_MsftSdk', variables.centOSStreamName) }}
+            isBuiltFromVmr: ${{ parameters.isBuiltFromVmr }}
+            vmrBranch: ${{ variables.VmrBranch }}
+            targetArchitecture: x64
+            pool: ${{ parameters.pool_Linux }}
+            container:
+              name: ${{ variables.centOSStreamContainerName }}
+              image: ${{ variables.centOSStreamContainerImage }}
+            buildSourceOnly: true
+            runOnline: true
+            runTests: false
+            extraProperties: "/p:EnableDevBuildAsDefaultForSourceOnly=true"
+
+        # Source Build Validation
+        - template: ../jobs/vmr-build.yml
+          parameters:
+            buildName: ${{ format('{0}_SourceBuild_Online_MsftSdk', variables.centOSStreamName) }}_Validation
+            isBuiltFromVmr: ${{ parameters.isBuiltFromVmr }}
+            vmrBranch: ${{ variables.VmrBranch }}
+            targetArchitecture: x64
+            pool: ${{ parameters.pool_Linux }}
+            container:
+              name: ${{ variables.centOSStreamContainerName }}
+              image: ${{ variables.centOSStreamContainerImage }}
+            buildSourceOnly: true
+            runOnline: true
+            runTests: true
+            skipBuild: true
+            extraProperties: >
+              /p:DotNetSourceOnlyTestOnly=true
+              /p:ExtraRestoreSourcePath=$(Pipeline.Workspace)/msft-pkgs
+              /p:SourceBuildTestsCustomSourceBuiltPackagesPath=$(Pipeline.Workspace)/msft-pkgs
+              /p:SourceBuildTestsCustomSourceBuiltPackagesPathIsInclusive=true
+            dependsOn:
+            - AzureLinux_x64_Cross_x64
+            - Browser_Shortstack_wasm
+            - Windows_x64
+            - ${{ format('{0}_SourceBuild_Online_MsftSdk_x64', variables.centOSStreamName) }}
+            testInitSteps:
```

Comment:
Will this pattern get duplicated when more legs are added?  Would it make sense to encapsulate this within a parameterized steps template?  The path could be passed as a parameter and then dynamically injected.  I'm not suggesting this be fixed now, but something to consider when more legs are migrated over.

Reply:
I'm not understanding how this could be factored out any more than it already is. I see this as being a pretty static set of steps. It won't change as we onboard new SB legs as the dependencies of all those legs on UB legs will be the same. I'm already going to be factoring out the SB and validation jobs into a separate template as was suggested here: https://github.com/dotnet/sdk/pull/48327#discussion_r2038145178. But the steps here, I'm not really seeing how things could be improved.

Reply:
Embedding this into a separate template will address my concern of having this copied for each SB leg.